### PR TITLE
Feature/#1307 loading loaders

### PR DESCRIPTION
**Pull request by ciaranschutte**
Loading States for cards.

Show loader if any parent loading a request.

Changed QueriesResolver to not make an empty query
i.e one with [] as request body, this is an invalid POST body
This solves the issue of resolvers making the queries with their default [] state

### CODE REVIEW

File: src/components/CohortBuilder/QueriesResolver.js [hunk 1]
```diff
@@ -32,28 +32,33 @@ class QueriesResolver extends React.Component {
...
+          () =>
+            new Promise(async resolve => {
+              const body = JSON.stringify(
+                queries.map(q => ({
+                  query: typeof q.query === 'string' ? q.query : print(q.query),
+                  variables: q.variables,
+                })),
+              );
+              try {
+                if (!useCache) {
```
Comment (hlminh2000) [inline src/components/CohortBuilder/QueriesResolver.js]: This was my hack, but `useCache` shouldn't be used here really, let's use its own `useLoadingState` prop based on our earlier chat?

File: src/components/CohortBuilder/Summary/index.js [hunk 2]
```diff
@@ -44,38 +43,45 @@ const Summary = ({
...
-                <DemographicChart data={demographicData} />
+                <DemographicChart data={demographicData} isParentLoading={isLoading} />
               </PaddedColumn>
               <PaddedColumn md={spacing.md} lg={spacing.lg}>
-                <SurvivalChart data={survivalPlotMock} />
+                <SurvivalChart data={survivalPlotMock} isParentLoading={isLoading} />
               </PaddedColumn>{' '}
               <PaddedColumn md={spacing.md} lg={spacing.lg}>
-                <AgeDiagChart data={ageDiagData} />
+                <AgeDiagChart data={ageDiagData} isParentLoading={isLoading} />
```
Comment (hlminh2000) [inline src/components/CohortBuilder/Summary/index.js]: Does it make more sense to expose `isParentLoading` as just `isLoading` instead? Totally just a naming thing, but I think it reads better :)
Each card can rename it internally to account for namespacing

Comment (ciaranschutte) [reply src/components/CohortBuilder/Summary/index.js]: Sure!

File: src/components/CohortBuilder/QueriesResolver.js (hunk 1 above)
Comment (ciaranschutte) [reply src/components/CohortBuilder/QueriesResolver.js]: @hlminh2000 
So I ended up only needing to check for parent loading state for the Cards and not having to add a new loading state to QueriesResolver, so useCache is just the same as before. Only change to QueriesResolver is the no query on empty array. Do you want me to just take out the useCache setstate?

**Comment by jberube** [comment]
```
Changed QueriesResolver to not make an empty query
i.e one with [] as request body, this is an invalid POST body
This solves the issue of resolvers making the queries with their default [] state
```
:+1:

File: src/components/CohortBuilder/Summary/index.js [hunk 3]
```diff
@@ -44,38 +43,45 @@ const Summary = ({
     ]}
   >
     {({ isLoading, data }) => {
```
Comment (jberube) [inline src/components/CohortBuilder/Summary/index.js]: if `data` can be `null`, it should either have a default here or on line 56.

**Pull request merged**